app/src/main/java/ui/ScheduleActivity.kt

```diff
@@ -372,7 +370,7 @@ class ScheduleActivity : DrawerActivity() {
             time.text = LocalTime.MIDNIGHT.withHour(hour).format(formatter)
 
             // Add it to the right container
-            timetableContainer!!.addView(timetableCell)
+            timetableContainer?.addView(timetableCell)
```

Comment:
For a lot of these methods with nullable view inputs, do they actually do anything when the view doesn't exist?
If not, add a null check at the beginning and skip all operations if null. Then deal with nonnull views afterwards.

If generating the data is always necessary, generate the data once, do a null check, then update the view if nonnull.

Reply:
The reason it's nullable in this situation is because of the different layouts for portrait and landscape modes. However, given that I've disabled landscape mode (for the time being), this is a non-issue. 
I intend on replacing this entire code with a library to generate the calendar view, as there are a few that do it way better than me. One of the things they can do that I don't is deal with courses that are at the same time.

Reply:
It's fine that it's nullable. What I'm saying is that if you are generating the data for the sole purpose of updating the view, you can skip the entire method if the views are null by adding a null check at the top